src/diff/props.js

```diff
@@ -96,7 +96,19 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 		&& !isSvg
 		&& (name in dom)
 	) {
-		dom[name] = value==null ? '' : value;
+		if (value==null) {
+			// This condition here guards against modifying a unonctrolled input's value
+			if (name!=='value' && name!=='checked') {
```

Comment:
ooooh crap I hadn't thought of this implication. There are a bunch more names here that would have to be disabled, I think this might be a sign that we need to try different overall solutions...

Reply:
Yup, you're right. I'm gonna start looking at it from other angles